qa/L0_lifecycle/lifecycle_test.py

```diff
@@ -2625,6 +2626,112 @@ def test_load_gpu_limit(self):
         except Exception as ex:
             self.assertTrue(False, "unexpected error {}".format(ex))
 
+    def test_concurrent_load_speedup(self):
+        # Initialize client
+        try:
+            triton_client = grpcclient.InferenceServerClient("localhost:8001",
+                                                             verbose=True)
+        except Exception as ex:
+            self.assertTrue(False, "unexpected error {}".format(ex))
+        # Load both models concurrently
+        model_names = ["identity_zero_1_int32_1", "identity_zero_1_int32_2"]
+        threads = []
+        for model_name in model_names:
+            threads.append(
+                threading.Thread(target=triton_client.load_model,
+                                 args=(model_name,)))
+        start_time = time.time()
+        for thread in threads:
+            thread.start()
+        for thread in threads:
+            thread.join()
+        end_time = time.time()
+        loading_time = end_time - start_time
+        # Each of the two models has a minimum loading delay of 10 seconds
+        # Speedup is observed when the concurrent loading time < 20 seconds but
+        # use a tighter bound of 15 seconds
+        self.assertLess(loading_time, 15.0,
+                        "Concurrent loading speedup not observed")
+        # Concurrent loading time cannot be < 10 seconds
+        self.assertGreaterEqual(loading_time, 10.0,
+                                "Invalid concurrent loading time")
+        # Make sure the models are loaded
+        self.assertTrue(triton_client.is_server_live())
+        self.assertTrue(triton_client.is_server_ready())
+        for model_name in model_names:
+            self.assertTrue(triton_client.is_model_ready(model_name))
+
+    def test_concurrent_load(self):
+        # Initialize client
+        try:
+            triton_client = grpcclient.InferenceServerClient("localhost:8001",
+                                                             verbose=True)
+        except Exception as ex:
+            self.assertTrue(False, "unexpected error {}".format(ex))
+        # Load same model concurrently
+        with concurrent.futures.ThreadPoolExecutor() as pool:
+            thread_1 = pool.submit(triton_client.load_model,
+                                   "identity_zero_1_int32")
+            time.sleep(2)  # wait between loads
+            thread_2 = pool.submit(triton_client.load_model,
```

Comment:
Should change model file so it produce different inference results, then perform second load. In this way we can check if the proper model is loaded and available.

Reply:
They are the exact same model which there is only one model file. This is to show the same model can be loaded concurrently, and both should succeed.

Do you mean put the same model name on two different namespaces?

Reply:
@kthui The second model load even though using the same model name should replace the model. I think you are missing that testing. I would suggest that you have 2 model files model_file_1 and model_file_2 in this test case..

Start with model_A with model_file_1, start loading model_A, replace model_file_1 with model_file_2 for model_A itself, then send request to load model_A again. 

You should be testing that finally loaded model_A is holding model_file_2.

Reply:
Thanks for clarifying.
Update commit: [Update concurrent test on same model name diff model file](https://github.com/triton-inference-server/server/pull/5360/commits/3825dd2a1a81da664afe73c36f9d20c151465e0c)